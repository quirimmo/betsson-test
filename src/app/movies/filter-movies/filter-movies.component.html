<form>
	<div class="row search-movie-container">
		<mat-form-field class="col-md-6 col-12 w-100">
			<input
				id="search-movie-name"
				[formControl]="nameText"
				matInput
				placeholder="Type the name of the movie..."
			/>
		</mat-form-field>
		<mat-form-field class="col-md-6 col-12 w-100">
			<mat-select
				id="seach-movie-genres"
				placeholder="Select the genres..."
				[formControl]="genresSelect"
				multiple
			>
				<mat-option *ngFor="let genre of genres" [value]="genre">{{
					genre
				}}</mat-option>
			</mat-select>
		</mat-form-field>
	</div>
</form>
