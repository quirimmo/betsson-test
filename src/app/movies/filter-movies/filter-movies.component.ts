import { Component, Input, OnInit, OnDestroy } from '@angular/core';
import { FormControl } from '@angular/forms';
import { Subscription } from 'rxjs';
import { ActivatedRoute, Router } from '@angular/router';
import { MoviesRouterService } from '../movies-router.service';

@Component({
	selector: 'filter-movies',
	inputs: ['genres'],
	styleUrls: ['./filter-movies.component.scss'],
	templateUrl: './filter-movies.component.html'
})
export class FilterMoviesComponent implements OnInit, OnDestroy {
	@Input()
	genres: string[];
	genresSelect = new FormControl();
	nameText = new FormControl();
	nameTextChangesSubscriber: Subscription;
	genresChangesSubscriber: Subscription;

	constructor(
		private moviesRouterService: MoviesRouterService,
		private route: ActivatedRoute,
		private router: Router
	) {}

	ngOnInit(): void {
		const searchName: string =
			this.moviesRouterService.getSearchNameParam() || '';
		const searchGenres: string[] =
			this.moviesRouterService.getSearchGenresParam() || this.genres;

		this.nameTextChangesSubscriber = this.nameText.valueChanges.subscribe(
			this.onChangeSearchName.bind(this)
		);
		this.genresChangesSubscriber = this.genresSelect.valueChanges.subscribe(
			this.onChangeSearchGenres.bind(this)
		);

		this.nameText.setValue(searchName);
		this.genresSelect.setValue(searchGenres);
	}

	ngOnDestroy(): void {
		this.nameTextChangesSubscriber.unsubscribe();
		this.genresChangesSubscriber.unsubscribe();
	}

	public onChangeSearchName(): void {
		this.moviesRouterService.appendSearchNameParam(this.nameText.value);
	}

	public onChangeSearchGenres(): void {
		this.moviesRouterService.appendSearchGenresParam(this.genresSelect.value);
	}
}
