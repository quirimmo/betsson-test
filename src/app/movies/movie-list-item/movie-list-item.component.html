<section class="movie-item-card">
	<div class="movie-item-card-title">{{ movie.name }}</div>
	<div>{{ movie.rate }}</div>
	<img mat-card-image [src]="'assets/' + movie.img" alt="{{ movie.name }}" />
	<div>{{ movie.genres.join(',') }}</div>
</section>
