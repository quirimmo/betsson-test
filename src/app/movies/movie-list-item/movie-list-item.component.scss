@import './../../../colors';
.movie-item-card {
	text-align: center;
	position: relative;
	margin-bottom: 30px;
	color: $blue;
}

.movie-item-card-title {
	min-height: 50px;
	font-size: 17pt;
}

.movie-item-card-rate {
	background-color: $yellow;
	color: $black;
	display: inline-block;
	top: -15px;
	padding: 2px 10px;
	position: relative;
	max-width: fit-content;
	margin: 0 auto;
}
