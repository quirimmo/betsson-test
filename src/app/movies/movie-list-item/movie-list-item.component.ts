import { Component, Input, OnInit } from '@angular/core';
import Movie from '../movie.model';

@Component({
	selector: 'movie-list-item',
	inputs: ['movie'],
	styleUrls: ['./movie-list-item.component.scss'],
	templateUrl: './movie-list-item.component.html'
})
export class MovieListItemComponent {
	@Input()
	movie: Movie;

	constructor() {}
}
