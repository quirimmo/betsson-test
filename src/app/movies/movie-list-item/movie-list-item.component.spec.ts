import { TestBed, async } from '@angular/core/testing';
import { FilterMoviesComponent } from '../filter-movies/filter-movies.component';
import { MovieListItemComponent } from '../movie-list-item/movie-list-item.component';

const providers = [];
const imports = [];
const declarations = [MovieListItemComponent];

describe('MoviesListItem Component', () => {
	let componentFixture, componentInstance;

	beforeEach(async(() => {
		TestBed.configureTestingModule({
			declarations,
			providers,
			imports
		}).compileComponents();
		componentFixture = TestBed.createComponent(MovieListItemComponent);
		componentInstance = componentFixture.debugElement.componentInstance;
	}));

	it('should create the component', () => {
		expect(componentFixture).toBeTruthy();
	});

	it('should match the snapshot', () => {
		expect(componentFixture).toMatchSnapshot();
	});
});
