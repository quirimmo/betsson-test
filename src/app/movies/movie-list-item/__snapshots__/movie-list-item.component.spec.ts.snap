// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MoviesListItem Component should match the snapshot 1`] = `
<movie-list-item>
  <section
    class="movie-item-card"
  >
    <div>
      <img
        class="movie-item-image"
        mat-card-image=""
      />
    </div>
    <div
      class="movie-item-card-rate"
    >
       Rate: 
      <strong>
        
      </strong>
    </div>
    <div
      class="movie-item-card-title"
    >
      
    </div>
  </section>
</movie-list-item>
`;
